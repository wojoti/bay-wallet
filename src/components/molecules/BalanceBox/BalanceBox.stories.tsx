import {ComponentMeta, ComponentStory} from '@storybook/react-native';

import BalanceBox from './BalanceBox';

export default {
  title: 'Molecules/BalanceBox',
  component: BalanceBox,
  argTypes: {onDetailsPress: {action: 'onDetailsPress'}},
} as ComponentMeta<typeof BalanceBox>;

const Template: ComponentStory<typeof BalanceBox> = args => (
  <BalanceBox {...args} />
);

export const Example = Template.bind({});
Example.args = {
  amount: '45.500,12',
  testId: '',
};
