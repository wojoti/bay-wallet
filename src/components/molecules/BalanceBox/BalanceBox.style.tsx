import {StyleSheet} from 'react-native';
import {balanceWhite, darkPurple} from 'theme/colors';

const style = StyleSheet.create({
  balanceBoxSizer: {
    width: '93%',
    marginTop: 35,
  },
  balanceBox: {
    alignItems: 'center',
    justifyContent: 'center',
  },
  backgroundBox: {
    width: '100%',
    height: 102,
    backgroundColor: darkPurple,
    borderRadius: 12,
    position: 'absolute',
  },
  whiteBox: {
    backgroundColor: balanceWhite,
    height: 140,
    width: '92%',
    borderRadius: 12,
    paddingVertical: 14,
    paddingLeft: 22,
    paddingRight: 18,
  },
  width: {
    width: 90,
  },
});
export default style;
