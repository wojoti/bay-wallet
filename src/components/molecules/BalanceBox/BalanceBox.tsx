import {
  Background,
  Button,
  Container,
  Header,
  Img,
  ImgType,
  Label,
} from 'components/atoms';
import {PropsWithChildren} from 'react';
import {darkGrey, green} from 'theme/colors';
import style from './BalanceBox.style';

export type BalanceBoxProps = PropsWithChildren<{
  onDetailsPress: () => void;
  testId?: string;
  amount: string;
  percent: string;
  balanceLabel: string;
  detailsLabel: string;
}>;

const BalanceBox = ({
  onDetailsPress,
  amount,
  percent,
  balanceLabel,
  detailsLabel,
  testId,
}: BalanceBoxProps) => {
  return (
    <Container customStyle={style.balanceBox} testId={testId}>
      <Container customStyle={style.backgroundBox} />
      <Container customStyle={style.whiteBox}>
        <Background
          source={require('img/wave.png')}
          customStyle={style.imgBackground}
          testId={'balancebox-background'}>
          <Container flex={1} justify="space-between">
            <Label size={15} color={darkGrey} testId={'balancebox-label'}>
              {balanceLabel}
            </Label>
            <Header size={28} testId={'balancebox-amount'}>
              $ {amount}
            </Header>
          </Container>
          <Container
            flex={1}
            align="flex-end"
            justify="space-between"
            direction="row">
            <Button
              customStyle={style.btnStyle}
              customFontStyle={style.btnTextStyle}
              onPress={onDetailsPress}
              testId={'balancebox-button'}>
              {detailsLabel}
            </Button>
            <Container
              customStyle={style.width}
              direction="row"
              align="flex-start"
              justify="space-between">
              <Header size={15} color={green} testId={'balancebox-percent'}>
                {percent} %
              </Header>
              <Img
                src={ImgType.waveGreen}
                height={'20'}
                width={41}
                alt={''}
                testId={'balancebox-wave'}
              />
            </Container>
          </Container>
        </Background>
      </Container>
    </Container>
  );
};

export default BalanceBox;
