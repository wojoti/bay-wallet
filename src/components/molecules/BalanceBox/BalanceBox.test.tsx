import '@testing-library/jest-native/extend-expect';
import {render, screen, within} from '@testing-library/react-native';
import TestRenderer from 'react-test-renderer';
import BalanceBox, {BalanceBoxProps} from './BalanceBox';

const mockOnDetailsPress = jest.fn();
const props: BalanceBoxProps = {
  onDetailsPress: mockOnDetailsPress,
  amount: '45.500,12',
  testId: 'test-balancebox-id',
  percent: '5',
  balanceLabel: Your main balance,
  detailsLabel: 'Details >',
};

test('should match snapshot', () => {
  const tree = TestRenderer.create(<BalanceBox {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('should render balancebox', () => {
  render(<BalanceBox {...props} />);
  const balanceboxElement = screen.getByTestId('test-balancebox-id');
  expect(balanceboxElement).toBeOnTheScreen();
});

test('should render balancebox - top welcoming label', () => {
  render(<BalanceBox {...props} />);
  const balanceboxElement = screen.getByTestId('test-balancebox-id');
  expect(balanceboxElement).toBeOnTheScreen();
  const balanceboxLabelElement =
    within(balanceboxElement).getByTestId('balancebox-label');
  expect(balanceboxLabelElement).toBeOnTheScreen();
  expect(balanceboxLabelElement).toBe('Your main balance');
});
