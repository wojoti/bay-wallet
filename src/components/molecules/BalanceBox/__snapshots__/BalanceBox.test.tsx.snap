// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<View
  style={
    [
      {
        "alignItems": "stretch",
        "flex": undefined,
        "flexDirection": "column",
        "justifyContent": "flex-start",
        "width": "100%",
      },
      {
        "alignItems": "center",
        "justifyContent": "center",
      },
    ]
  }
  testID="test-balancebox-id"
>
  <View
    style={
      [
        {
          "alignItems": "stretch",
          "flex": undefined,
          "flexDirection": "column",
          "justifyContent": "flex-start",
          "width": "100%",
        },
        {
          "backgroundColor": "#26204E",
          "borderRadius": 12,
          "height": 102,
          "position": "absolute",
          "width": "100%",
        },
      ]
    }
  />
  <View
    style={
      [
        {
          "alignItems": "stretch",
          "flex": undefined,
          "flexDirection": "column",
          "justifyContent": "flex-start",
          "width": "100%",
        },
        {
          "backgroundColor": "#F8F8FF",
          "borderRadius": 12,
          "height": 140,
          "width": "92%",
        },
      ]
    }
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        [
          {
            "flex": 1,
          },
          {
            "flex": 1,
            "paddingLeft": 22,
            "paddingRight": 18,
            "paddingVertical": 14,
          },
        ]
      }
    >
      <Image
        source={
          {
            "testUri": "../../../src/img/wave.png",
          }
        }
        style={
          [
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            {
              "height": undefined,
              "width": undefined,
            },
            undefined,
          ]
        }
      />
      <View
        style={
          [
            {
              "alignItems": "stretch",
              "flex": 1,
              "flexDirection": "column",
              "justifyContent": "space-between",
              "width": "100%",
            },
            undefined,
          ]
        }
      >
        <Text
          style={
            {
              "color": "#595959",
              "fontFamily": "Poppins-Light",
              "fontSize": 15,
              "fontWeight": "400",
              "textAlign": "left",
            }
          }
        >
          Your main balance
        </Text>
        <Text
          style={
            {
              "color": "black",
              "fontFamily": "Poppins-Bold",
              "fontSize": 28,
              "textAlign": "left",
            }
          }
        >
          $ 
          45.500,12
        </Text>
      </View>
      <View
        style={
          [
            {
              "alignItems": "flex-end",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "space-between",
              "width": "100%",
            },
            undefined,
          ]
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": false,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "alignItems": "center",
                "borderRadius": 8,
                "borderStyle": "solid",
                "borderWidth": 1,
                "height": 65,
                "justifyContent": "center",
                "opacity": 1,
              },
              false,
              {
                "borderWidth": 0,
                "color": "transparent",
                "height": 30,
              },
            ]
          }
          testID="balancebox-button"
        >
          <Text
            style={
              [
                {
                  "fontFamily": "Poppins-Bold",
                  "fontSize": 18,
                  "textAlign": "center",
                },
                {
                  "color": "#363062",
                  "fontSize": 15,
                },
              ]
            }
            testID="button-text"
          >
            Details &gt;
          </Text>
        </View>
        <View
          style={
            [
              {
                "alignItems": "flex-start",
                "flex": undefined,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "width": "100%",
              },
              {
                "width": 90,
              },
            ]
          }
        >
          <Text
            style={
              {
                "color": "#61C277",
                "fontFamily": "Poppins-Bold",
                "fontSize": 15,
                "textAlign": "left",
              }
            }
          >
            5
             %
          </Text>
          <SvgMock
            height="20"
            width={41}
          />
        </View>
      </View>
    </View>
  </View>
</View>
`;
